modules/sdk-coin-sol/src/sol.ts

```diff
@@ -574,16 +580,35 @@ export class Sol extends BaseCoin {
 
     // Build the transaction
     const MPC = await EDDSAMethods.getInitializedMpcInstance();
-    const accountId = MPC.deriveUnhardened(bitgoKey, `m/0`).slice(0, 64);
-    const bs58EncodedPublicKey = new SolKeyPair({ pub: accountId }).getAddress();
+    let bs58EncodedPublicKey;
+    let balance;
+    let addressIndex;
+    const feePerSignature = await this.getFees();
+    const totalFee = params.durableNonce ? feePerSignature * 2 : feePerSignature;
+
+    // Check for first derived wallet with funds
+    for (let i = startingDerivationIndex; i < endingDerivationIndex; i++) {
+      const derivationPath = `m/${i}`;
+      const accountId = MPC.deriveUnhardened(bitgoKey, derivationPath).slice(0, 64);
+      bs58EncodedPublicKey = new SolKeyPair({ pub: accountId }).getAddress();
+
+      balance = await this.getAccountBalance(bs58EncodedPublicKey);
+
+      if (balance > totalFee) {
+        addressIndex = i;
+        break;
+      }
+    }
+
+    if (balance < totalFee) {
```

Comment:
Could we test how this would work with unexpected inputs? What if the endingDerivationIndex is zero (we might never have that case, but still) - then wouldn't balance always be undefined at this point since the for loop never triggers, and perhaps the comparison would error out?